docs/source/hallway.rst

```diff
@@ -0,0 +1,363 @@
+(inheriting from the graphenv ``Vertex``) and ``HallwayModel``
+(inheriting from the graphenv ``Model``). We then provide a working
+example of building and running a hallway environment.
+
+HallwayState
+------------
+
+(See ``graphenv.examples.hallway.hallway_state`` for the full
+implementation).
+
+The ``HallwayState`` represents all of the problem logic at the level of
+a single vertex that enables graphenv to automate the overarching search
+and RL training. This class inherits from ``graphenv.vertex.Vertex``
+which has a number of required methods and attributes which we step
+through below.
+
+``__init__``
+~~~~~~~~~~~~
+
+As you’d expect, problem configuration happens here. The hallway state
+is completely specified by the current and end positions,
+
+.. code:: python
+
+   def __init__(
+           self,
+           corridor_length: int,
+           cur_pos: int = 0,
+       ) -> None:
+           """Initializes this HallwayState.
+           Args:
+               corridor_length (int): length of the vertex chain
+               cur_pos (int, optional): initial vertex index. Defaults to 0.
+           """
+           super().__init__()
+           self.end_pos = corridor_length
+           self.cur_pos = cur_pos
+
+``observation_space``
+~~~~~~~~~~~~~~~~~~~~~
+
+Returns a ``gym.spaces.Space`` object that describes the structure of
+the data used to represent a vertex. In the hallway problem,
+
+.. code:: python
+
+   @property
+   def observation_space(self) -> gym.spaces.Dict:
+       """HallwayStates are observed with a dictionary containing a single
+       key, 'cur_pos', with an integer value between 0 and self.end_pos,
+       indicating the index of the vertex.
+       Returns:
+           gym.spaces.Dict: The observation space for HallwayStates.
+       """
+       return gym.spaces.Dict(
+           {
+               "cur_pos": gym.spaces.Box(
+                   low=np.array([0]), high=np.array([self.end_pos]), dtype=int
+               ),
+           }
+       )
+
+where ``cur_pos`` is the integer index of the current position. The box
+space has a single element containing the index of the current position
+but, in general, can contain multiple, complex subspaces.
+
+``new``
+~~~~~~~
+
+Returns a new vertex instance with updated state. For the hallway
+problem,
+
+.. code:: python
+
+   def new(self, cur_pos: int):
+       """Convenience function for duplicating the existing node.
+       Returns:
+           HallwayState : a copy of this HallwayState.
+       """
+       return HallwayState(self.end_pos, cur_pos)
+
+where ``cur_pos`` is the position of the new state.
+
+``root``
+~~~~~~~~
+
+Returns the root vertex. In the hallway problem, we always go back to
+state 0,
+
+.. code:: python
+
+   @property
+   def root(self) -> "HallwayState":
+       """
+       Returns:
+           HallwayState: initial state (vertex at index 0)
+       """
+       return self.new(0)
```

Comment:
could probably skip these methods to get the the more illustrative `reward` and `children` methods.

Reply:
ok, thx for the feedback -- i erred on the side of putting too much detail here -- will remove